### PR TITLE
refactor: project maintenance

### PR DESCRIPTION
**Pull request by mcmxcdev**
#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/HEAD/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/HEAD/CODE_OF_CONDUCT.md) that this project adheres to.

#### Description of Change
- remove redundant `@adonisjs/auth`, `@adonisjs/session` and `@emotion/react` dependencies
- clean up vscode settings
- remove redundant `launch.json` and `tasks.json`
- clean up some TS and ESLint issues
- migrate from deprecated "husky install" to "husky"

#### Motivation and Context
Fixup improvement possibilities that I came across

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`pnpm prepare-code`)
- [x] `pnpm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: .vscode/launch.json
Comment (mcmxcdev) [inline .vscode/launch.json:1]: Seemed like a pointless file. If anyone is using this I can restore it.

File: .vscode/settings.json
Comment (mcmxcdev) [inline .vscode/settings.json:1]: In my opinion, only a minimum of settings should be set here.

A lot of the ones that I have removed are very opinionated and should be set in VSCode user-specific settings, not be applied for every user of the project.

File: .vscode/tasks.json
Comment (mcmxcdev) [inline .vscode/tasks.json:1]: Seemed like a pointless file. If anyone is using this I can restore it.

Comment (vraravam) [reply .vscode/tasks.json:1]: i agree - dont know why this was committed in

**Review by vraravam — APPROVED** [review]

**Comment by vraravam** [comment]
@mcmxcdev  - I would suggest that we completely remove the `.vscode` folder and add it to the `.gitignore` file.

As a contributor, in the future, I would not want the `.latestNewFangledEditorConfig` to show up and disrupt my local setup with something that a new person committed from their choice of editor.
As a common baseline, imo, that's what we have the editor config file which is supposed to be a defacto standard for.

**Comment by mcmxcdev** [comment]
> @mcmxcdev - I would suggest that we completely remove the `.vscode` folder and add it to the `.gitignore` file.
> 
> As a contributor, in the future, I would not want the `.latestNewFangledEditorConfig` to show up and disrupt my local setup with something that a new person committed from their choice of editor. As a common baseline, imo, that's what we have the editor config file which is supposed to be a defacto standard for.

Okay sounds good to me, removed the folder completely in [`141251d` (#1682)](https://github.com/ferdium/ferdium-app/pull/1682/commits/141251d4567699cda98343551bf4523827d62374)

**Review by vraravam — APPROVED** [review]
lgtm

**Pull request merged**